package.json

```diff
@@ -1,45 +1,47 @@
 {
-  "name": "@metrics/guard",
-  "version": "1.0.1",
-  "description": "Module to guard against excessive metric permutation creation in a metric stream",
-  "main": "lib/guard.js",
-  "files": [
-    "lib"
-  ],
-  "scripts": {
-    "bench": "node benchmark/benchmark.js",
-    "test": "tap test/*.js",
-    "test:coverage": "tap test/*.js --cov",
-    "lint": "eslint .",
-    "lint:fix": "eslint . --fix"
-  },
-  "author": "Trygve Lie <post@trygve-lie.com>",
-  "license": "MIT",
-  "keywords": [
-    "server-statistics",
-    "server-stats",
-    "statistics",
-    "metrics",
-    "monitoring",
-    "guard"
-  ],
-  "dependencies": {
-    "readable-stream": "^3.4.0"
-  },
-  "devDependencies": {
-    "eslint": "6.8.0",
-    "eslint-config-airbnb-base": "14.0.0",
-    "eslint-config-prettier": "6.10.0",
-    "eslint-plugin-import": "2.20.1",
-    "eslint-plugin-prettier": "3.1.2",
-    "@metrics/client": "2.5.0",
-    "@metrics/metric": "2.3.1",
-    "fastbench": "1.0.1",
-    "prettier": "1.19.1",
-    "tap": "14.10.6"
-  },
-  "repository": {
-    "type": "git",
-    "url": "https://github.com/metrics-js/client.git"
-  }
+    "name": "@metrics/guard",
+    "version": "1.0.1",
+    "description": "Module to guard against excessive metric permutation creation in a metric stream",
+    "main": "lib/guard.js",
+    "types": "guard.d.ts",
+    "files": [
+        "lib",
+        "guard.d.ts"
+    ],
+    "scripts": {
+        "bench": "node benchmark/benchmark.js",
+        "test": "tap test/*.js",
+        "test:coverage": "tap test/*.js --cov",
+        "lint": "eslint .",
+        "lint:fix": "eslint . --fix"
+    },
+    "author": "Trygve Lie <post@trygve-lie.com>",
+    "license": "MIT",
+    "keywords": [
+        "server-statistics",
+        "server-stats",
+        "statistics",
+        "metrics",
+        "monitoring",
+        "guard"
+    ],
+    "dependencies": {
+        "readable-stream": "^3.4.0",
+        "@metrics/metric": "^2.3.1"
```

Comment:
Adding types forces us to make a dev dep into a dep?
This could end up with the downstream project having several versions of the metrics module pulled in etc. Is there any way around this?

Reply:
I think so, given the `import` on L2 of https://github.com/metrics-js/guard/pull/35/files#diff-176edfab6a9cd295dc5ae8a4501ec0bc6850e164824aea857aad24d65dc7a07c

We could make it a peerDependency, but that won't really solve anything if other dependencies are pinned to an older version. In any case we should indicate for our TS users the minimum expected version of the `metric` package.

Isn't Metrics JS designed with this multi-client version scenario in mind, since there's multiple clients but only the one consumer?

Reply:
> Isn't Metrics JS designed with this multi-client version scenario in mind, since there's multiple clients but only the one consumer?

One of the ideas is that there can be multiple produces and multiple consumers. The one thing which should not change (much) is the Metric Object flowing through the different modules. Ex; a producer should emit an instance of the Metric Class and a consumer should receive one but would normally change its character to push to the system the consumer is intended to push metrics too. Between the producers and consumers one can hook any tool in the chain to do stuff with the metrics but they should always receive and emit object of the Metric Class.

Anyhow; these are streams and any object, string or binary data can flow through them (these are always in object mode though). TS does nada to prevent what flows through here in production so adding the Metric Class as a full dependency when its only used in dev / build time feels very wrong to me.

Though; If we add runtime "type" check on the objects passed in to the streams I'm fine adding the Metric Class as a dependency.

Reply:
> as a full dependency when its only used in dev / build time

I agree, but it's used by our users in their dev/build time, so it's a way to ensure they have what they need both for development and runtime.

Anyway, I've moved the dependency back down to `devDependencies`. My thinking is that the chances are high that `@metrics/metric` gets pulled in by for instance the client. If not, I suppose folks will have to install it directly themselves.